components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImpl.java

```diff
@@ -268,4 +364,70 @@ private Action updateEndpointAuthenticationProperties(String actionType, String
+        ACTION_VALIDATOR.isValidActionName(action.getName());
+        ACTION_VALIDATOR.isValidEndpointUri(action.getEndpoint().getUri());
+        Authentication endpointAuthentication = action.getEndpoint().getAuthentication();
+        doEndpointAuthenticationValidation(endpointAuthentication);
+    }
+
+    /**
+     * Perform pre validations on action model when updating an existing action.
+     * This is specifically used during HTTP PATCH operation and
+     * only validate non-null and non-empty fields.
+     *
+     * @param action Action update model.
+     * @throws ActionMgtClientException if action model is invalid.
+     */
+    private void doPreUpdateActionValidations(Action action) throws ActionMgtClientException {
+
+        if (action.getName() != null) {
+            ACTION_VALIDATOR.isValidActionName(action.getName());
+        }
+        if (action.getEndpoint() != null && action.getEndpoint().getUri() != null) {
+            ACTION_VALIDATOR.isValidEndpointUri(action.getEndpoint().getUri());
+        }
+        if (action.getEndpoint() != null && action.getEndpoint().getAuthentication() != null) {
+            Authentication endpointAuthentication = action.getEndpoint().getAuthentication();
+            doEndpointAuthenticationValidation(endpointAuthentication);
+        }
+    }
+
+    /**
+     * Perform pre validations on endpoint authentication model.
+     *
+     * @param authentication Endpoint authentication model.
+     * @throws ActionMgtClientException if endpoint authentication model is invalid.
+     */
+    private void doEndpointAuthenticationValidation(Authentication authentication)
+            throws ActionMgtClientException {
+
+        String authenticationType = authentication.getType().getName();
+        ACTION_VALIDATOR.isBlank(ENDPOINT_AUTHENTICATION_TYPE_FIELD, authenticationType);
+        if (authenticationType.equals(Authentication.Type.BASIC.getName())) {
+            ACTION_VALIDATOR.isBlank(USERNAME_FIELD,
+                    authentication.getProperty(Authentication.Property.USERNAME).getValue());
+            ACTION_VALIDATOR.isBlank(PASSWORD_FIELD,
+                    authentication.getProperty(Authentication.Property.PASSWORD).getValue());
+        } else if (authenticationType.equals(Authentication.Type.BEARER.getName())) {
+            ACTION_VALIDATOR.isBlank(ACCESS_TOKEN_FIELD,
+                    authentication.getProperty(Authentication.Property.ACCESS_TOKEN).getValue());
+        } else if (authenticationType.equals(Authentication.Type.API_KEY.getName())) {
+            String apiKeyHeader = authentication.getProperty(Authentication.Property.HEADER).getValue();
+            ACTION_VALIDATOR.isBlank(API_KEY_HEADER_FIELD, apiKeyHeader);
+            ACTION_VALIDATOR.isValidHeader(apiKeyHeader);
+            ACTION_VALIDATOR.isBlank(API_KEY_VALUE_FIELD,
+                    authentication.getProperty(Authentication.Property.VALUE).getValue());
+        }
+    }
```

Comment:
Let's add more tests to `ActionManagementServiceImplTest` test the newly added lines

Reply:
This is updated